### PR TITLE
TandemFoil: transfer AirfRANS golden config WD=1e-2 + gc=1.0

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis
The AirfRANS golden config (WD=1e-2 + grad-clip=1.0 + T_max=5) delivered a cumulative -22.3% improvement (PR #2727). Weight decay regularizes the model to avoid overfitting on training geometries, and gradient clipping stabilizes training dynamics. TandemFoil uses the Lion optimizer (which has different gradient scale characteristics than AdamW), but WD=1e-2 + gc=1.0 as a combination should still provide beneficial regularization. This transfers the entire regularization configuration from AirfRANS to TandemFoil.

## Instructions
Run the TandemFoil baseline with AirfRANS golden regularization config:

```bash
SENPAI_TIMEOUT_MINUTES=180 SENPAI_MAX_EPOCHS=9999 python train.py \
  --dataset tandemfoil \
  --optimizer lion \
  --lr 2e-4 \
  --cosine-t-max 10 \
  --no-use-ema \
  --model-slices 64 \
  --enable-fourier \
  --enable-te-coord-frame \
  --enable-cp-panel \
  --enable-cp-panel-tandem-only \
  --asinh-pressure \
  --residual-prediction \
  --enable-pressure-prior-addition \
  --weight-decay 1e-2 \
  --grad-clip 1.0 \
  --wandb_group tandem-lr2e4-golden \
  --epochs 9999
```

Changes vs baseline:
- `--weight-decay 1e-2` (added, was unset/default)
- `--grad-clip 1.0` (added, was unset/default)

## Baseline
Current best TandemFoil metrics (PR #2610, W&B: ixs1rqgk):
- val_primary/surface_pressure_mae: **45.07** (lower is better)

Reproduce baseline:
```bash
cd target/icml2026 && python train.py --dataset tandemfoil --optimizer lion --lr 2e-4 --cosine-t-max 10 --no-use-ema --model-slices 64 --enable-fourier --enable-te-coord-frame --enable-cp-panel --enable-cp-panel-tandem-only --asinh-pressure --residual-prediction --enable-pressure-prior-addition --epochs 999
```

### CODE REVIEW

**Comment by tcapelle** [comment]
STUDENT mikasa:

## Results

### Primary Metrics

| Metric | Baseline (PR #2610) | This Run | Change |
|--------|---------------------|----------|--------|
| val_primary/surface_pressure_mae | 45.07 | **31.98** | **-29.0%** |
| test_primary/surface_pressure_mae | — | **33.16** | — |

### Per-Split Test MAE

| Split | surface_pressure_mae |
|-------|---------------------|
| single_in_dist | 33.89 |
| geom_camber_rc | 40.40 |
| geom_camber_cruise | 33.27 |
| re_rand | 25.08 |

### Per-Split Val MAE (epoch 159)

| Split | surface_pressure_mae |
|-------|---------------------|
| single_in_dist | 33.64 |
| geom_camber_rc | 43.32 |
| geom_camber_cruise | 17.77 |
| re_rand | 33.20 |

### Training Details

- **Epochs:** 159 (180-min timeout, best at **final epoch** — still improving!)
- **W&B run:** 6na5rnil
- **Peak VRAM:** ~58.5 GB (GPU 0)
- **Final loss:** 0.054
- **Grad clip evolution:** 750/750 clipped at epoch 1 (norm=10.04) → 12/750 at epoch 159 (norm=0.32)

### Command

```bash
cd target/icml2026 && CUDA_VISIBLE_DEVICES=0 SENPAI_TIMEOUT_MINUTES=180 SENPAI_MAX_EPOCHS=9999 python train.py \
  --dataset tandemfoil --optimizer lion --lr 2e-4 --cosine-t-max 10 \
  --no-use-ema --model-slices 64 --enable-fourier --enable-te-coord-frame \
  --enable-cp-panel --enable-cp-panel-tandem-only --asinh-pressure \
  --residual-prediction --enable-pressure-prior-addition \
  --weight-decay 1e-2 --grad-clip 1.0 \
  --wandb-group tandem-lr2e4-golden --epochs 9999 \
  --agent mikasa --wandb-name "mikasa/tandem-wd1e2-gc10"
```

### What Happened

The AirfRANS golden regularization config (WD=1e-2 + gc=1.0) transfers powerfully to TandemFoil with the Lion optimizer. **29% improvement** over baseline, with the best metric at the very last epoch — strong evidence of continued headroom.

Key observations:
1. **Grad clipping was critical early on** — 100% of steps clipped at epoch 1 with mean grad norm 10.04. This suggests the baseline (no clipping) was experiencing gradient spikes that degraded training.
2. **Grad norms settled naturally** — by epoch 159, only 1.6% of steps were clipped (norm=0.32). The WD regularization smoothed the loss landscape so clipping becomes less needed over time.
3. **Consistent downward envelope** — no plateau. The trajectory from 176→105→67→47→42→40→37→34→32 shows steady improvement across 159 epochs.
4. **All splits improved uniformly** — cruise split (17.77 val) is particularly strong, while geom_camber_rc (43.32) has the most remaining headroom.
5. **Note:** Infinity in test_geom_camber_cruise/mae_vol_p — likely a data artifact (this appears in other TandemFoil runs too).

### Suggested Follow-ups

1. **Longer training** — Best at epoch 159 with no convergence. A 360-min run could push below 30.
2. **gc=1.5 on TandemFoil** — AirfRANS showed gc=1.5 > gc=1.0 (PR #2737). TandemFoil may also benefit from slightly looser clipping now that WD is providing regularization.
3. **5L/256d + WD+gc** — The deeper architecture (PR #2595) was baseline 52.81 without regularization. Adding WD+gc to 5L/256d could unlock a further jump.
4. **WD sweep** — WD=1e-2 was transferred from AirfRANS with AdamW. Lion handles weight decay differently (decoupled). WD=5e-3 or WD=2e-2 might be better tuned for Lion's gradient scaling.

**Comment by tcapelle** [comment]
ADVISOR: Closing PR #2777 because ADVISOR: Closing — your result of 31.98 does not beat the new 30.10 baseline. Thanks for the effort, mikasa. We are moving on to stronger configurations that combine the gc=0.5 insight with optimized learning rates..
